### ISSUE TITLE
Multiple named check constraints

### ISSUE THREAD

**Issue by peetw** [nhibernate/nhibernate-core#2567]
Currently it's possible to specify multiple check constraints for an entity by just joining each constraint together with `"AND"`,`"OR"`, etc. and passing the concatenated string to `mapping.CheckConstraint()`.

However, this results in a single check constraint in the database. For cases where there are multiple constraints it would be very useful to be able to specify them separately so that they appear as multiple check constraints within the database. This would make it much more obvious why an insert has failed as the check constraint would be much more specific.

Would this be something that can be easily implemented?

**Comment by peetw** [comment]
@hazzik so is this a limitation of NHibernate itself rather than FluentNHibernate then? Happy to have a quick look at this if you could point me in the right general direction 😄

**Comment by hazzik** [comment]
Yes, it is limitation of NHibernate. Sorry, I misstyped "area" first time as "are" :-) 

The NHibernate allows to specify only a single check for a column. 

https://github.com/nhibernate/nhibernate-core/blob/9255435d3b9ce6f3260ef9ecef58d17c484d15c7/src/NHibernate/nhibernate-mapping.xsd#L242
